pkg/apis/extensions/validation/validation.go

```diff
@@ -279,7 +282,7 @@ func ValidateDeploymentStrategy(strategy *extensions.DeploymentStrategy, fldPath
 		if strategy.RollingUpdate == nil {
 			allErrs = append(allErrs, field.Required(fldPath.Child("rollingUpdate"), "this should be defaulted and never be nil"))
 		} else {
-			allErrs = append(allErrs, ValidateRollingUpdateDeployment(strategy.RollingUpdate, fldPath.Child("rollingUpdate"))...)
+			allErrs = append(allErrs, ValidateRollingUpdateDeployment(strategy.RollingUpdate, replicas, fldPath.Child("rollingUpdate"))...)
```

Comment:
@kargakis We need to update this before validation, right? the location I can find is `PrepareForCreate` and `PrepareForUpdate` in the [strategy.go](https://github.com/kubernetes/kubernetes/blob/9ae2dfacf196ca7dbee798ee9c3e1663a5f39473/pkg/registry/extensions/deployment/strategy.go#L62), is it the right location?